notebooks/working/3.0-mab-independent-variable-simulation.md

```diff
@@ -34,487 +47,269 @@ import random
+    fitter_object.fit()
+    # Get the best distribution and store in dictionary
+    BestDict = fitter_object.get_best()
+    cont_dict[ind_var]['distribution'] = list(BestDict.items())[0][0]
+    cont_dict[ind_var]['parameters'] = list(BestDict.items())[0][1]
+    return cont_dict
+
+## Function for looping around individual specific variables
+def indSpecDist(data_long, obs_id_col, ind_spec, var_types):
+    ind_spec_dict = defaultdict(dict)
+    for ind_var in ind_spec:
+        # generate array of values for individual specific variable
+        var_val = pd.Series([(data_long.loc[data_long[obs_id_col] == x][ind_var].unique()[0]) for x in data_long[obs_id_col].unique()])
+        var_type = var_types[ind_var]
+        # Get distribution if variable is categorical
+        if isCategorical(var_type):
+            # If only one category
+            if isUnique(var_val):
+                ind_var_dict = getConstantDist(ind_var, var_val)
+            # If more than one category
+            else:
+                ind_var_dict = getCategoricalDist(ind_var, var_val)
+        else:
+            ind_var_dict = getContinuousDist(ind_var, var_val, cont_dists)
+        ind_spec_dict.update(ind_var_dict)
+    return ind_spec_dict
+
+## Functions to replace code for getting
+## distributions for alternative specific
+## variables
+
+def getConstantDistAlt(alt_var, alt_data, alt_name_dic, alt):
+    constant_dict = defaultdict(dict)
+    # Add name of alternative to variable and store distriburion & parameters
+    constant_dict[alt_var+'_'+alt_name_dic[alt]]['distribution'] = 'constant'
+    constant_dict[alt_var+'_'+alt_name_dic[alt]]['parameters'] = alt_data[alt_var].unique()
+    return constant_dict
+
+def getCategoricalDistAlt(alt_var, alt_data, alt_name_dic, alt):
+    categ_dict = defaultdict(dict)
+    categ_dict[alt_var+'_'+alt_name_dic[alt]]['distribution'] = 'categorical'
+    np_array_range = np.arange(alt_data[alt_var].max()+1)
+    array_bincount = np.bincount(alt_data[alt_var])
+    probs = array_bincount / len(alt_data[alt_var])
+    categ_dict[alt_var+'_'+alt_name_dic[alt]]['parameters'] = [np_array_range,
+                                                               probs]
+    return categ_dict
+
+def getContinuousDistAlt(alt_var, alt_data, alt_name_dic, alt, cont_dists):
+    cont_dict = defaultdict(dict)
+    # Use the Fitter library to fit distributions
+    # to the data
+    fitter_object = Fitter(data=alt_data[alt_var],
+                           distributions=cont_dists,
+                           timeout=30)
+    fitter_object.fit()
+    # Get the best distribution and store in dictionary
+    BestDict = fitter_object.get_best()
+    # Add name of alternative to variable and store distriburion & parameters
+    cont_dict[alt_var+'_'+alt_name_dic[alt]]['distribution'] = list(BestDict.items())[0][0]
+    cont_dict[alt_var+'_'+alt_name_dic[alt]]['parameters'] = list(BestDict.items())[0][1]
```

Comment:
Same comment as above re the logic for creating the alternative specific variable name.

Reply:
Yes sir!